src/modules/logger/logger.cpp

```diff
@@ -1070,6 +1107,25 @@ void Logger::run()
 
 #endif /* DBGPRINT */
 
+		} else { // not logging
+
+			// try to subscribe to new topics, even if we don't log, so that:
+			// - we avoid subscribing to many topics at once, when logging starts
+			// - we'll get the data immediately once we start logging (no need to wait for the next subscribe timeout)
+			if (next_subscribe_topic_index != -1) {
+				for (uint8_t instance = 0; instance < ORB_MULTI_MAX_INSTANCES; instance++) {
```

Comment:
using uint8_t as index is not an optimization, you should use a counter of processor word size

Reply:
I know, I copy-pasted that :/
Fixed in 37902d99de04a329c8